reasoner/resolution/resolver/ConcludableResolver.java

```diff
@@ -103,22 +104,28 @@ protected void receiveAnswer(Response.Answer fromDownstream, int iteration) {
         if (!responseProducer.hasProduced(conceptMap)) {
             responseProducer.recordProduced(conceptMap);
 
-            // update partial derivation provided from upstream to carry derivations sideways
-            ResolutionAnswer.Derivation derivation = new ResolutionAnswer.Derivation(map(pair(fromDownstream.sourceRequest().receiver(),
-                                                                                              fromDownstream.answer())));
-            assert fromUpstream.answerBounds().isMapped();
+            ResolutionAnswer.Derivation derivation;
+            if (explanations()) {
+                // update partial derivation provided from upstream to carry derivations sideways
+                derivation = new ResolutionAnswer.Derivation(map(pair(fromDownstream.sourceRequest().receiver(),
+                                                                      fromDownstream.answer())));
+            } else {
+                derivation = null;
+            }
+
             ResolutionAnswer answer = new ResolutionAnswer(fromUpstream.answerBounds().asMapped().aggregateToUpstream(conceptMap),
                                                            concludable.toString(), derivation, self(), fromDownstream.answer().isInferred());
 
             respondToUpstream(new Response.Answer(fromUpstream, answer), iteration);
         } else {
-            ResolutionAnswer.Derivation derivation = new ResolutionAnswer.Derivation(map(pair(fromDownstream.sourceRequest().receiver(),
-                                                                                              fromDownstream.answer())));
-            ResolutionAnswer deduplicated = new ResolutionAnswer(fromDownstream.answer().derived(), concludable.toString(),
-                                                                 derivation, self(), fromDownstream.answer().isInferred());
-            LOG.trace("{}: Recording deduplicated answer derivation: {}", name(), deduplicated);
-            resolutionRecorder.tell(actor -> actor.record(deduplicated));
-
+            if (explanations()) {
+                ResolutionAnswer.Derivation derivation = new ResolutionAnswer.Derivation(map(pair(fromDownstream.sourceRequest().receiver(),
+                                                                                                  fromDownstream.answer())));
+                ResolutionAnswer deduplicated = new ResolutionAnswer(fromDownstream.answer().derived(), concludable.toString(),
+                                                                     derivation, self(), fromDownstream.answer().isInferred());
+                LOG.trace("{}: Recording deduplicated answer derivation: {}", name(), deduplicated);
+                resolutionRecorder.tell(actor -> actor.record(deduplicated));
+            }
```

Comment:
Resolved on the basis that right now large derivations will be carried in the Answers unless a flag is used to prevent this. We'd like to revisit this architecture later when we re-enable explanations.